### PR TITLE
Collections: Adding a requirements.yml file

### PR DESCRIPTION
**Pull request by Spredzy**
##### SUMMARY

Specify a list of collections to rely upon

##### ISSUE TYPE

 - Feature Pull Request
 
##### COMPONENT NAME

 - Installer

##### AWX VERSION

- devel

##### ADDITIONAL INFORMATION

- N/A

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -210,7 +211,10 @@ requirements_awx: virtualenv_awx
 requirements_awx_dev:
 	$(VENV_BASE)/awx/bin/pip install -r requirements/requirements_dev.txt
 
-requirements: requirements_ansible requirements_awx
+requirements_collections:
+	ansible-galaxy collection install -r requirements/collections_requirements.yml -p ${COLLECTION_BASE}
+
+requirements: requirements_ansible requirements_awx requirements_collections
```
Comment (AlanCoding) [inline Makefile:218]: I don't know what exactly uses this target, like, what deployment methods share it.

`COLLECTION_BASE ?= /collections` seems like kind of a weird location for the folder. Myself, I would not hesitate to put this in the `/var/lib/awx/...` location

Comment (AlanCoding) [reply Makefile:218]: I also don't see this invoked in the Dockerfile for the installer here or in the docker-compose setup.

This command can take a frustrating long period of time, and it certainly seems right to do it at the time of image building.

Comment (Spredzy) [reply Makefile:218]: > I don't know what exactly uses this target, like, what deployment methods share it.

Packaging.

> COLLECTION_BASE ?= /collections seems like kind of a weird location for the folder.

This is just a default value for variable. This is later on overridden where appropriate to point to other path.

@AlanCoding The PR submitted from last night was and still is a draft meant to show a working flow with package install. From an upstream perspective the changes to the various Dockerfiles have been added this morning.

Comment (Spredzy) [reply Makefile:218]: Also, I am still looking for guidance on where we would preferably have this laid down. Given `/var/lib/awx/collections` is not an option. Do we want to make it clear we are vendoring it by having it in `/var/lib/awx/vendor/collections` ?

Comment (AlanCoding) [reply Makefile:218]: > Packaging.

but it's messier than that, right?

```
$ grep -r "make requirements" installer/
installer//roles/image_build/templates/Dockerfile.j2:RUN cd /tmp && VENV_BASE="/var/lib/awx/venv" make requirements_awx requirements_ansible_py3
```

https://github.com/ansible/awx/blob/eafd40291e786599c712b36bc4012199619a97fc/installer/roles/image_build/templates/Dockerfile.j2#L71

I think maybe we would add `make requirements_collections` there

**Comment by AlanCoding** [comment]
In terms of plans, I want to very soonish combine this PR with https://github.com/ansible/awx/pull/6232 so functionality can be tested.

File: Makefile (hunk 1 above)
Comment (Spredzy) [reply Makefile:218]: @AlanCoding its already there -> https://github.com/ansible/awx/pull/6264/files#diff-ea72c08e2b13ba6d1d9334f604684fe2R97-R98

Comment (wenottingham) [reply Makefile:218]: /var/lib/awx/vendor/inventory_collections ?

**Comment by AlanCoding** [comment]
Closing in favor of https://github.com/ansible/awx/pull/6283

**Pull request closed without merging**